code/modules/species/station/xenomorph_hybrids/hybrid_abilities.dm

```diff
@@ -9,3 +9,30 @@
 		if(O)
 			O.color = "#422649"
 	return
+
+/datum/ability/species/xenomorph_hybrid
+	action_icon = 'icons/screen/actions/xenomorph.dmi'
+
+/datum/ability/species/xenomorph_hybrid/regenerate
+	name = "Rest and regenerate"
+	desc = "Lie down and regenerate your health"
+	action_state = "regenerate"
+	windup = 0 SECOND
+	interact_type = ABILITY_INTERACT_TRIGGER
+	always_bind = TRUE
+	ability_check_flags = ABILITY_CHECK_RESTING
+	mobility_check_flags = MOBILITY_IS_CONSCIOUS
+
+/datum/ability/species/xenomorph_hybrid/regenerate/on_trigger()
+	. = ..()
+	var/mob/living/carbon/human/O = owner
+	if(istype(O))
+		to_chat(O, SPAN_NOTICEALIEN("We begin to mend our wounds."))
+		O.active_regen = TRUE
```

Comment:
i feel like the entire logic of "can we regen" could be handled in a single method that checks the relevant things such as nutrition and if they are laying down

you could then call this here before setting active_regen to true, and do the same in /datum/species/xenohybrid/handle_environment_special(var/mob/living/carbon/human/H)